rgsync/redis_gears_write_behind.py

```diff
@@ -426,3 +458,77 @@ def __init__(self, GB, keysPrefix, mappings, connector, name, version=None):
         filter(TryWriteToTarget(self)).\
         foreach(UpdateHash).\
         register(mode='sync', prefix='%s*' % keysPrefix, eventTypes=['hset', 'hmset'], convertToStr=False)
+
+
+class RGJSONWriteBehind(RGWriteBase):
+    # JSONWrite Behind
+    # The big deal is that:
+    # 1. It calls ValidateJSONHash instead of ValidateHash
+    # 2. The init requires a data key, in order to go through the sub map and update
+    #    within the JSON document.
+    def __init__(self, GB, keysPrefix, connector, name, version=None,
+                 onFailedRetryInterval=5, batch=100, duration=100,
+                 eventTypes=['json.set', 'json.del',
+                             'json.strappend', 'json.arrinsert', 'json.arrappend',
+                             'json.arrtrim', 'json.arrpop'],
+                 dataKey="gears"):
+
+        mappings = {'redis_data': 'gears'}
```

Comment:
So we are still using redis_data and not mapping everything?

Reply:
Unfortunately due to how things work under the covers in mongo, I needed a top level key. There wasn't a choice.

Reply:
But you can have the user specify the top-level key only, and map the entire json to this key and not only what under redis_data, no?

Reply:
We could - but IMHO I don't see the point. This is ephemeral data and on the mongo side, not present.